src/main/java/org/folio/marc4ld/service/marc2ld/Marc2BibframeMapperImpl.java

```diff
@@ -65,15 +66,30 @@ public Resource fromMarcJson(String marc) {
       return null;
     }
     var reader = getReader(marc);
-    var instance = new Resource().addType(INSTANCE);
+    var instanceAndWork = createInstanceAndWork();
     while (reader.hasNext()) {
-      fillInstanceFields(reader.next(), instance);
+      fillInstanceFields(reader.next(), instanceAndWork.instance);
     }
+    setLabelsAndIds(instanceAndWork);
```

Comment:
Compute the id of Work and Instance here.
(By this time, all the properties of both Work and Instance will be populated)